src/Common/RWLock.h

```diff
@@ -90,6 +90,7 @@ class RWLockImpl : public std::enable_shared_from_this<RWLockImpl>
 
     RWLockImpl() = default;
     void unlock(GroupsContainer::iterator group_it, const String & query_id) noexcept;
-    void dropOwnerGroupAndPassOwnership(GroupsContainer::iterator group_it) noexcept;
+    /// @param next - notify next after begin, used on writer lock failures
+    void dropOwnerGroupAndPassOwnership(GroupsContainer::iterator group_it, bool next) noexcept;
```

Comment:
maybe use a default argument?
```suggestion
    void dropOwnerGroupAndPassOwnership(GroupsContainer::iterator group_it, bool next = false) noexcept;
```

Reply:
Default arguments is error prone, and especially in such major place (there are not too much callers by the way).